## 62% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 2
Expected to fail tests: 1
Passed tests: 5
```
```
IDs of failed tests: 703133,727229 
```
```
IDs of expected to fail tests: 727232 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=cl/fix-typed-chain-issue-again&PR_ID=20943&APK_NAME=20943.apk&TR_CASE_IDS=703133,727229">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633396">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `show-profiles` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:243: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8090a83dac7a4fe59f482081e10afffe?auth=fb7dd9c363d717be3895c35ec81c5273#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633397">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='container']/android.widget.TextView[@text='Ether']/../android.widget.TextView[3]` </div><div>Device 2: `Text` is `0.02649 ETH` </div></blockquote></p>

```
critical/test_wallet.py:159: in test_wallet_send_eth
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.4675 but expected to be 0.4676
```

<br/><br/></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633400">test_wallet_add_remove_watch_only_account</a>, id: 727232 </b></td></tr><tr><td><p><blockquote><div>Device 1: `Text` is `0x8d2413447ff297d30bdc475f6d5cb00254685aae` </div><div>Device 1: Click system back button </div></blockquote></p>

```
critical/test_wallet.py:274: in test_wallet_add_remove_watch_only_account
    self.home_view.driver.fail(
base_test_case.py:179: in fail
    pytest.fail('Device %s: %s' % (self.number, text))
 Device 1: Incorrect address '0x8d2413447ff297d30bdc475f6d5cb00254685aae' is shown when swiping between accounts, expected one is '0x8d2413447ff297d30bdc475f6d5cb00254685aae' 
```

<b>[[Missing networks in account address, https://github.com/status-im/status-mobile/issues/20166]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/afaa200e88db4d87b5ba6caac10dbb56?auth=f38d83ec632dd3e792a97b30618ae070#100">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633399">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/afaa200e88db4d87b5ba6caac10dbb56?auth=f38d83ec632dd3e792a97b30618ae070#36">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633394">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/253ca640e6254492967df205f4251e48?auth=a3ed8d725ba069d5756faefd01ecccb1#93">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b21b45ea52cb46b28b220582fca20f21?auth=1b0de26ff787b4958321d02d07c8d151#82">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633393">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8090a83dac7a4fe59f482081e10afffe?auth=fb7dd9c363d717be3895c35ec81c5273#50">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633395">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/19989ea3c2bb447a92cdd00f348a1e5e?auth=e20e73e47ec9dd2736437e3df0f0887b#156">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/933d792344a7434ea63ceed4d6d91b53?auth=70527173061e9fce069c111714d76834#104">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3633398">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/wallet_connect/events.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/events.cljs
@@ -265,9 +265,7 @@
                              persisted-sessions)]
      (when (seq expired-sessions)
        (log/info "Updating WalletConnect persisted sessions due to expired/inactive sessions"
-                 {:expired   expired-sessions
-                  :persisted persisted-sessions
-                  :active    sessions}))
+                 {:expired expired-sessions}))
      {:fx (mapv (fn [{:keys [topic]}]
                   [:dispatch [:wallet-connect/disconnect-session topic]])
                 expired-sessions)

--- a/src/status_im/contexts/wallet/wallet_connect/processing_events.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/processing_events.cljs
@@ -126,40 +126,37 @@
 (rf/reg-event-fx
  :wallet-connect/process-sign-typed
  (fn [{:keys [db]}]
-   (let [[address raw-data] (wallet-connect-core/get-db-current-request-params db)
-         parsed-raw-data    (transforms/js-parse raw-data)
-         session-chain-id   (-> (wallet-connect-core/get-db-current-request-event db)
-                                (get-in [:params :chainId])
-                                wallet-connect-core/eip155->chain-id
-                                str)
-         data-chain-id      (-> parsed-raw-data
-                                transforms/js->clj
-                                signing/typed-data-chain-id
-                                str)
-         parsed-data        (try (-> parsed-raw-data
-                                     (transforms/js-dissoc :types :primaryType)
-                                     (transforms/js-stringify 2))
-                                 (catch js/Error _ nil))]
-     (cond
-       (nil? parsed-data)
+   (try
+     (let [[address raw-data] (wallet-connect-core/get-db-current-request-params db)
+           parsed-raw-data    (transforms/js-parse raw-data)
+           session-chain-id   (-> (wallet-connect-core/get-db-current-request-event db)
+                                  (get-in [:params :chainId])
+                                  wallet-connect-core/eip155->chain-id)
+           data-chain-id      (-> parsed-raw-data
+                                  transforms/js->clj
+                                  signing/typed-data-chain-id)
+           parsed-data        (-> parsed-raw-data
+                                  (transforms/js-dissoc :types :primaryType)
+                                  (transforms/js-stringify 2))]
+       (if (and data-chain-id
+                (not= session-chain-id data-chain-id))
+         {:fx [[:dispatch
+                [:wallet-connect/wrong-typed-data-chain-id
+                 {:expected-chain-id session-chain-id
+                  :wrong-chain-id    data-chain-id}]]]}
+         {:db (update-in db
+                         [:wallet-connect/current-request]
+                         assoc
+                         :address      (string/lower-case address)
+                         :display-data (or parsed-data raw-data)
+                         :raw-data     raw-data)
+          :fx [[:dispatch [:wallet-connect/show-request-modal]]]}))
+     (catch js/Error err
        {:fx [[:dispatch
               [:wallet-connect/on-processing-error
-               (ex-info "Failed to parse JSON typed data" {:data raw-data})]]]}
-
-       (and (seq session-chain-id) (not= session-chain-id data-chain-id))
-       {:fx [[:dispatch
-              [:wallet-connect/wrong-typed-data-chain-id
-               {:expected-chain-id session-chain-id
-                :wrong-chain-id    data-chain-id}]]]}
-
-       :else
-       {:db (update-in db
-                       [:wallet-connect/current-request]
-                       assoc
-                       :address      (string/lower-case address)
-                       :display-data (or parsed-data raw-data)
-                       :raw-data     raw-data)
-        :fx [[:dispatch [:wallet-connect/show-request-modal]]]}))))
+               (ex-info "Failed to parse JSON typed data"
+                        {:error err
+                         :data  (wallet-connect-core/get-db-current-request-params db)})]]]}))))
 
 (rf/reg-event-fx
  :wallet-connect/wrong-typed-data-chain-id
@@ -171,7 +168,9 @@
                                    wallet-connect-core/chain-id->network-details
                                    :full-name)
          toast-message         (i18n/label :t/wallet-connect-typed-data-wrong-chain-id-warning
-                                           {:wrong-chain    wrong-network-name
+                                           {:wrong-chain    (or wrong-network-name
+                                                                (wallet-connect-core/chain-id->eip155
+                                                                 wrong-chain-id))
                                             :expected-chain expected-network-name})]
      {:fx [[:dispatch
             [:toasts/upsert

--- a/src/status_im/contexts/wallet/wallet_connect/signing.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/signing.cljs
@@ -4,6 +4,7 @@
             [status-im.contexts.wallet.wallet-connect.core :as core]
             [status-im.contexts.wallet.wallet-connect.rpc :as rpc]
             [utils.hex :as hex]
+            [utils.number :as number]
             [utils.transforms :as transforms]))
 
 (defn typed-data-chain-id
@@ -16,7 +17,8 @@
                             (some #(= "chainId" (:name %))))
         data-chain-id  (-> typed-data
                            :domain
-                           :chainId)]
+                           :chainId
+                           number/parse-int)]
     (when chain-id-type?
       data-chain-id)))
 

